### ISSUE TITLE
Have a more advanced strategy to enable/disable DocumentProvider

### ISSUE THREAD

**Issue by gsmet** [quarkusio/quarkus#1882]
We disabled `DocumentProvider` which deals with `org.w3c.dom.Document` entirely in Quarkus as, with the default provider resolution, as soon as XML is produced/consumed, it embarks a significant amount of code in the native image, even if not used at all.

A more advanced strategy based on the return type/parameter types of the endpoint methods would probably be nice for this particular provider.

Something sufficiently generic so that we could add other providers similar to that would be nice.

Basically, we need a map of providers => list of types triggering them. These providers should be ignored in the resolution except if a type triggers them in the return type/parameters. I think we should also still considers the `@Produces`/`@Consumes` annotations.

See `ResteasyCommonConfig`, `DeleteDocumentProvider`.

**Comment by sberyozkin** [comment]
@gsmet, can you clarify please, why `DeleteDocumentProvider` is needed, instead of 'manually' excluding this provider in `ResteasyCommonProcessor.categorizeProviders` ? thanks

**Comment by gsmet** [comment]
To be more precise, the issue is that as soon as we do XML, we enable `DocumentProvider`, even if we don't deal with `Document` at all.

**Comment by gsmet** [comment]
I'm closing this one for now. If someone encounters the issue, please open a new issue and we will deal with it.
